File: Devices/ECG/message_handler.cpp
#include <iostream>
#include <sstream>
#include "PatientWrapper.hpp"
#include <vector>
#include "MQTTAsync.h"

extern Patient original_patient;
extern Patient temporary_patient;

vector<string> split(const char* data)
{
    vector<string> contents;
    stringstream ss(data);
    string token;
    while (getline(ss, token, ';')) {
        contents.push_back(token);
    }
    return contents;
}

void handle_incoming_messages(void *context, char *topicName, int topicLen, MQTTAsync_message *message)
{
    vector<string> contents = split((char*)message->payload);
    original_patient.setId(contents[0]);
    original_patient.setName(contents[1]);
    original_patient.setGender(contents[2]);
    original_patient.setAge(stoi(contents[3]));
    original_patient.setProcedureName(contents[4]);
    original_patient.setConsumables(contents[5]);
    original_patient.setReportIds(contents[6]);
    original_patient.setBusyStatus(stoi(contents[7]));
    temporary_patient = original_patient;
    MQTTAsync_freeMessage(&message);
    MQTTAsync_free(topicName);
}

File: Devices/PythonClient/message_handler.py
import settings
import copy


class MessageHandler:
    @staticmethod
    def handle_edit():
        condition = settings.temporary_patient.getId() != settings.original_patient.getId()        
        if condition:
            settings.update_patient_count()
            settings.update_inventory(settings.original_patient.getProcedureName().lower().strip())
            settings.send_mail_if_stock_low()
            
        
    @staticmethod
    def handle_message(message):
        contents = list(message.strip().split(";"))
        settings.original_patient.setId(contents[0])
        settings.original_patient.setName(contents[1])
        settings.original_patient.setGender(contents[2])
        settings.original_patient.setAge(int(contents[3]))
        settings.original_patient.setProcedureName(contents[4])
        settings.original_patient.setConsumables(contents[5])
        settings.original_patient.setReportIds(contents[6])
        settings.original_patient.setBusyStatus(int(contents[7]))

        MessageHandler.handle_edit()

        settings.temporary_patient = copy.deepcopy(settings.original_patient)


        
